src/main/scala/nodes/nlp/NGramsHashingTF.scala

```diff
@@ -0,0 +1,117 @@
+package nodes.nlp
+
+import java.lang.Integer.{rotateLeft => rotl}
+
+import breeze.linalg.SparseVector
+import workflow.Transformer
+
+import scala.collection.mutable
+
+/**
+ * Converts the n-grams of a sequence of terms to a sparse vector representing their frequencies,
+ * using the hashing trick: https://en.wikipedia.org/wiki/Feature_hashing
+ *
+ * It computes a rolling MurmurHash3 instead of fully constructing the n-grams, making
+ * it more efficient than using [[NGramsFeaturizer]] followed by [[HashingTF]], although
+ * it should return the exact same feature vector.
+ *
+ * Individual terms are hashed using Scala's `.##` method. We may want to convert to MurmurHash3 for strings,
```

Comment:
Yeah, the terms' hash values are combined together using MurmurHash3 to form the n-gram hashes, but the individual terms are still hashed using the `.##` method.